exchange/src/exchange/tickers/engine/order_storage.hpp

```diff
@@ -33,16 +33,39 @@ struct StoredOrder {
     }
 
     StoredOrder(
-        const std::unique_ptr<manager::GenericTrader>& trader, SIDE side,
-        std::string ticker, double quantity, double price, uint64_t tick
+        std::shared_ptr<manager::GenericTrader> trader, SIDE side, std::string ticker,
+        double quantity, double price, uint64_t tick
     ) :
-        trader(trader),
+        trader(std::move(trader)),
         ticker(std::move(ticker)), side(side), price(price), quantity(quantity),
         tick(tick), order_index(get_and_increment_global_index())
     {}
 
-    explicit StoredOrder(messages::MarketOrder&& other, uint64_t tick) :
+    StoredOrder(StoredOrder&& other) noexcept :
+        trader(std::move(other.trader)), ticker(std::move(other.ticker)),
+        side(other.side), price(other.price), quantity(other.quantity),
+        tick(other.tick), order_index(other.order_index)
+    {}
+
+    StoredOrder&
+    operator=(StoredOrder&& other) noexcept
```

Comment:
first of all this is implemented incorrectly

second of all, this is the same as the default. just ` = default;` and be done